### DISCUSSION TITLE
Why is "metadata/document-ids" represented in schema as an array

### DISCUSSION THREAD

**Discussion by SParekh** [usnistgov/OSCAL#929] [Q&A] [answered]
As per the issue #864 and [Model review document](https://pages.nist.gov/OSCAL/contribute/model-review/document-id.pdf), "document-id" _"Provides identifier for the class(or “true identity”) of document."_

Questions:
1. As per the documentation if  “document-Id” provides identifier for the class(or “true identity”) of document, why is it an array and not an attribute ?
https://github.com/usnistgov/OSCAL/blob/91084cdad37e88a5fcaf05e5a80c3a81e72a62c5/src/metaschema/oscal_metadata_metaschema.xml#L33

2. Is NIST authored Frameworks/Baselines going to be updated to represent the class of the document as part of Metadata ? This would be very useful for systems that are leveraging these documents to get this identifying information from "document" itself instead of relying on "filenames" for the same.

**Comment by david-waltermire** [comment, marked as answer]
> As per the documentation if “document-Id” provides identifier for the class(or “true identity”) of document, why is it an array and not an attribute ?

A given document may have multiple document identifiers. This can be due to a duplicate identifier assignment within the same system (e.g., old id/new id), or to provide document identifier in different systems (e.g., organization-specific series, DOI). For these reasons the document-id is an array to allow multiple document-ids to be provided.

> Is NIST authored Frameworks/Baselines going to be updated to represent the class of the document as part of Metadata ? This would be very useful for systems that are leveraging these documents to get this identifying information from "document" itself instead of relying on "filenames" for the same.

We are currently thinking about an approach to do this. We are also interested in producing a tutorial which outlines some best-practices in this area. We will update the OSCAL content once we have a way forward.
